lib/src/utils/provisioning.dart

```diff
@@ -262,6 +262,7 @@ Future<ProvisionedMeshNode> _provisioning(
     await bleMeshManager.refreshDeviceCache();
     await bleMeshManager.disconnect();
     _cancelProvisioningCallbackSubscription(bleMeshManager);
+    provisionedMeshNode.nodeName = deviceToProvision.name;
```

Comment:
I feel like it should not be done here or not like this. Can we try to do it earlier? For example, when the UnprovisionedMeshNode object is created ? Or maybe expose the parameter name in provisioning method..🤔